aiidalab/app.py

```diff
@@ -49,6 +51,8 @@
 
 logger = logging.getLogger(__name__)
 
+Dependency = namedtuple("Dependency", ["installed", "required"])
```

Comment:
Just come back to this, I think this may not best, if we make this a class and add the method or property `satisfied` which check the core dependencies compatibility directly here will things easier in other places where this class is used. The type hint can also more easily applied to the vairables. The question is whether the `_CORE_PACKAGE` can be defined in the `Dependency`?

Reply:
Indeed, `dataclass` might be a more powerful approach. I would keep the definition of core packages separate though, since that seems logically different.

Reply:
It would be something like:
```python
# class Denpendency is used to store the information of the dependency and check the compatibility of the dependency
# wrt to the core dependencies defined in _CORE_DEPENDENCIES
# the class has two attributes:
# - installed: type of Package which is the package installed for the dependency
# - required: type of Requirement which is the requirement for the dependency
# the property strictly_satisfied return True if compatible with the core dependencies
class Dependency:
    
    _CORE_DEPENDENCIES = ["aiida-core"]
    def __init__(self, installed: Package, required: Requirement):
        self.installed = installed
        self.required = required

    @property
    def strictly_satisfied(self):
        # check the installed version wrt to the _CORE_DEPENDENCIES
        if self.installed.name in self._CORE_DEPENDENCIES:
            return self.installed.version in self.required
        else:
            return self.installed.version in self.required or self.installed.version > self.required
```

I need to go now, let's discuss tomorrow and see if we want to go far in this.

Reply:
Nice. I think with dataclass decorator you should be able to simplify like this

```python
@dataclass(frozen=True)
class Dependency:
    _CORE_DEPENDENCIES = ["aiida-core"]
    installed: str
    required: str

    @property
    def strictly_satisfied(self):
.
.
.
```

Reply:
@yakutovicha just pointing out this discussion above since you're now working on this.

Reply:
I ended up replacing the whole class with a dictionary `{"installed":.., "required":..}` in [65f83cd](https://github.com/aiidalab/aiidalab/pull/347/commits/65f83cde6eed1d601c3ad39a4f428f728f9c0e9c). I don't think there is a reason to keep it since the versions with non-matching core dependencies are not shown.